### PR TITLE
169 resource loader jar

### LINKED ISSUE #169: Resources not loaded in JAR

### PR DESCRIPTION
**Pull request by mpsijm**
Will (finally!) close #169 

I have created a util class `Resource` that handles the loading of resources.
The pom had to be adapted to add `src/main/resources` to the `target/test-classes` folder. This was why the builds kept failing every time, Travis couldn't find the main resources anymore.

Good luck reviewing, shouldn't be a tough read :)

<review, configuration

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/audio/SoundEffectPlayer.java [hunk 1]
```diff
@@ -45,17 +42,11 @@ public SoundEffectPlayer(final String soundPath) {
     @Override
     protected final void initializeAudio(final String source) {
         try {
-            audioClip = new AudioClip(new File(source).toURI().toURL()
-                    .toString());
-        } catch (MalformedURLException mue) {
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/audio/SoundEffectPlayer.java]: Does the MediaException already cover the MalformedURLException?

<review

**Comment by mdingjan** [comment]
I went through your changes and checked them with our SAT. Your changes are looking fine.
Besides, I didn't find any SAT violations.
I'm happy that you finally achieved to solve your resources problem with the JARs. :)

Please answer my question and after that, this branch is ready for merging!

<review

File: src/main/java/nl/tudelft/ti2206/group9/audio/SoundEffectPlayer.java (hunk 1 above)
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/audio/SoundEffectPlayer.java]: No, the `MalformedURLException` is handled in the class `Resource`. In the new code, `Resource.getURIString(source)` is called, which handles any malformed URLs.
The `MediaException` can still be thrown by the AudioClip, and is thus still being caught here in `SoundEffectPlayer`.

<review

Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/audio/SoundEffectPlayer.java]: I see, thanks for your response! :)

File: src/main/java/nl/tudelft/ti2206/group9/util/Resource.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+     */
+    public static String getURIString(final String packagePath) {
+        return getURI(packagePath).toString();
+    }
+
+    /**
+     * @param packagePath The package path of the resource.
+     * @return the URL.toString() of this resource.
+     */
+    public static String getURLString(final String packagePath) {
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/util/Resource.java]: Method is never used according to my IDE.
Is this true? If not, please remove.

< review, logic

**Comment by thervh70** [comment]
Please come up with some tests in order to make sure the exceptions are thrown when needed. The test cases do not have to be complex, but we have to keep our coverage high, and our code quality as well!

This branch is looking good.

**Comment by mpsijm** [comment]
@thervh70 That exception will be fixed in #174b. This has nothing to do with Resources.
I have resolved your other comments though :)

<review,testing

**Comment by MKerkhof** [comment]
Audio seems okay now, but sometimes when the speeding up is occurring, there is some kind of weird lag. Any idea why?

Furthermore, code looks good and works good :) 

<Review

**Comment by MathiasMeuleman** [comment]
At my branch the sound is still hideous. Did I miss something?

<review

**Comment by mpsijm** [comment]
@MathiasMeuleman: Use `mvn clean` to remove the target folder, so the resources get copied again. The timestamp didn't change, so Maven doesn't overwrite them.

<review, configuration

**Comment by MathiasMeuleman** [comment]
Tnx, that works fine now! Maybe fix for the laggy speedup is for the next iteration. This works correct now, will merge.

<review

**Pull request merged**